src/board.rs

```diff
@@ -142,6 +142,136 @@ impl Chessboard {
         println!();
     }
 
+    /// Converts a square index (0-63) to its corresponding chess rank and file.
+    ///
+    /// # Arguments
+    ///
+    /// * `square` - The square index (0-63) on the chessboard.
+    ///
+    /// # Returns
+    ///
+    /// A tuple containing the corresponding file (character) and rank (usize) for the given square.
```

Comment:
It may seem nice  to include (character) and (usize) here, but this is what's called a `magic number` so we should remove them.